konrad/core.py

```diff
@@ -228,8 +238,28 @@ def is_converged(self):
         Returns:
             bool: ``True`` if converged, else ``False``.
         """
-        # TODO: Implement proper convergence criterion (e.g. include TOA).
-        return np.all(np.abs(self.deltaT) < self.delta)
+        
+        self.newDN = self.newN - self.oldN
+        self.newDDN = np.abs(self.newDN) - np.abs(self.oldDN)
+        
+        test1 = np.abs(self.newDN) <= self.delta
+        test2 = np.abs(self.newDDN) <= (self.delta / 100)
+        
+        self.oldDN = self.newDN
+        self.oldDDN = self.newDDN
+        
+        if ( test1 and test2 ):
+            self.counteq += 1
+        else:
+            if self.counteq > 0:
+                self.counteq -= 1
+            else:
+                self.counteq = 0
```

Comment:
As far as I understand, you want to check, if the convergence criterion was fulfilled for 2000 iterations.
Two comments on that:
1. Please add a brief comment that explains this :)
2. Would it be possible to calculate this based on, e.g. `30.0 // self.timestep_days' to account for changing time steps?

Reply:
1. I will put a comment in.

2. Yes, it is something that I will try.

Reply:
@lkluft, I am thinking in point number two and I would like your opinion: Instead of a threshold of a number of time steps in the quasi-equilibrium, why not a time interval in the quasi-equilibrium:

```python
    ...
    # I redefine self.counteq as a time delta in the initialisation
    # The same for self.post_count
    self.counteq = datetime.timedelta(microseconds=0)
    self.post_count = utils.parse_fraction_of_day(post_count)
    ...
    # Inside is_converged() method
    ...
    # If both test1 and test2 are true increments the counts in equilibrium
    # in one. In any other case it reduces the counts in one or maintains
    # them at zero
    if ( test1 and test2 ):
        self.counteq += self.timestep
    else:
        if self.counteq > 0:
            self.counteq -= self.timestep
        else:
            self.counteq = datetime.timedelta(microseconds=0)
    
    # If the equilibrium are larger than the threshold counts, it declares
    # convergence
    if self.counteq > self.post_count:
        return True
    else:
        return False
```

Reply:
Yes, I support this approach. That's what I was aiming at with my comment :) 

BTW: You can simplify the last line to `return self.counteq > self.post_count` ;)